python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -213,6 +246,24 @@ def _task_done_callback(actor_to_return):
                 lambda: _task_done_callback(actor_to_return),
             )
 
+    def _refresh_actor_cls(self):
+        """When `self._ray_remote_args_fn` is specified, this method should
+        be called prior to initializing the new worker in order to get new
+        remote args passed to the worker. It updates `self.cls` with the same
+        `_MapWorker` class, but with the new remote args from
+        `self._ray_remote_args_fn`."""
+        assert self._ray_remote_args_fn, "_ray_remote_args_fn must be provided"
+        remote_args = self._ray_remote_args.copy()
+        new_remote_args = self._ray_remote_args_fn()
+
+        # Override args from user-defined remote args function.
+        new_and_overriden_remote_args = {}
```

Comment:
we always override with the param from `ray_remote_args_fn` because Ray Data will use default parameters for `scheduling_strategy` and insert into `ray_remote_args`. We want to always override scheduling strategies provided by the user defined function, and IMO we should do the same for other parameters if the user provides `ray_remote_args_fn`.